### PR TITLE
fix(loader): support v1.6.3+ (and other fixes)

### LINKED ISSUE #307: angular 1.6.3 breaks batarang
Enabling batarang while working with angular 1.6.3 totally breaks the application to watch.
I don't know why, it's working fine in 1.6.2. I use angular via cdn but i don't think the problem comes from that.

Could someone investigate, if project is still alive? thanks.

Here is what chrome's console outputs:

> Uncaught TypeError: ac(...).info is not a function
>     at Ce (angular.js:2793)
>     at angular.js:33194
>     at angular.js:6
> Ce @ angular.js:2793
> (anonymous) @ angular.js:33194
> (anonymous) @ angular.js:6
> 
> angular-sanitize.js:555Uncaught TypeError: f.module(...).provider(...).info is not a function
>     at angular-sanitize.js:555
>     at angular-sanitize.js:6
> (anonymous) @ angular-sanitize.js:555
> (anonymous) @ angular-sanitize.js:6
> 
> angular-messages.js:267Uncaught TypeError: l.module(...).info is not a function
>     at angular-messages.js:267
>     at angular-messages.js:6
> (anonymous) @ angular-messages.js:267
> (anonymous) @ angular-messages.js:6
> 
> angular-cookies.js:24Uncaught TypeError: c.module(...).info is not a function
>     at angular-cookies.js:24
>     at angular-cookies.js:6
> (anonymous) @ angular-cookies.js:24
> (anonymous) @ angular-cookies.js:6
> 
> angular.js:38Uncaught Error: [$injector:modulerr] http://errors.angularjs.org/1.6.3/$injector/modulerr?p0=ng&p1=Error%3A%20%5…udflare.com%2Fajax%2Flibs%2Fangular.js%2F1.6.3%2Fangular.min.js%3A22%3A277)
>     at angular.js:38
>     at angular.js:4920
>     at p (angular.js:402)
>     at g (angular.js:4880)
>     at eb (angular.js:4802)
>     at c (angular.js:1909)
>     at Object.Pc.ea.resumeBootstrap (angular.js:1938)
>     at maybeBootstrap (hint.js:569)

### PR DESCRIPTION
**Pull request by gkalpak**
- Updated dependencies and configs.
- Switched to yarn.
- Added support for the `angular.Module#info()` method, necessary in v1.6.3+.

Fixes #307.

### CODE REVIEW

File: dist/hint.js [hunk 1]
```diff
@@ -19,23 +19,23 @@ angular.hint.onAny(function (data, severity) {
 
 },{"./loader.js":2,"angular-hint":3}],2:[function(require,module,exports){
 /**
- * @license AngularJS v1.3.0-build.3042+sha.76e57a7
- * (c) 2010-2014 Google, Inc. http://angularjs.org
+ * @license AngularJS v1.6.4-build.5322+sha.d96e58f
+ * (c) 2010-2017 Google, Inc. http://angularjs.org
```
Comment (petebacondarwin) [inline dist/hint.js]: Is this stuff just taken straight from the 1.6.4 loader or is it modified?

**Comment by Narretz** [comment]
The hint.js is copied from angular-hint. :)

File: dist/hint.js (hunk 1 above)
Comment (gkalpak) [inline dist/hint.js]: It is modified in a couple of places (e.g. `minErr` doesn't have access to `toDebugString()` so it uses a simpler implementation and there are a couple of other changes (marked with `ANGULAR HINT ALTERATION` comments) to support 1.2.x as well.

Maybe it is better to remove the header and leave a comment that it is based on v1.6.4 loader.js?

Comment (gkalpak) [inline dist/hint.js]: (BTW, reviewing with `?w=1` reduces the noise significantly :smiley:)

**Comment by gkalpak** [comment]
Yes, `hint.js` is generated based on `angular-hint`'s hint.js (as published on npm). I copied the latest `loader.js` from angular/agular.js and tweaked as necessary.

Batarang has some tests, but afaict they are about the DevTools panel, rather than how the extension interacts with AngularJS apps. So, stuff like that would not be caught.

File: dist/hint.js (hunk 1 above)
Comment (gkalpak) [inline dist/hint.js]: I just realized the original comment was about `dist/hint.js`. My responses were about `loader.js`.
`dist/hint.js` includes `loader.js` (among othr things).

I had accidentally commited `dist/hint.js` and the `loader.js` changes on different commits (although the `dist/hinst.js` changes are directly affected by the `loader.js` changes). I have now amended squashed the two commits into one. Sorry for the confusion :grin:

**Comment by gkalpak** [comment]
Merged :grin: (Happy to address additional feedback/comments in follow-up PRs.)

**Pull request closed without merging**